### PR TITLE
Allow only triggering captcha after hitting threshold of failed attempts

### PR DESCRIPTION
**Pull request by nekodex**
Add `RECAPTCHA_THRESHOLD=n` to `.env` to trigger after `n` failed attempts. (if n = 0, it will always show the captcha)

depends on:
- [x] #6526

### CODE REVIEW

**Review by nanaya — CHANGES_REQUESTED** [review]
Conflicts =)

**Comment by peppy** [comment]
@nanaya is this good to go minus conflicts?

**Comment by nanaya** [comment]
I haven't tested it

**Review by nanaya — CHANGES_REQUESTED** [review]

File: resources/views/layout/_popup_login.blade.php [hunk 1]
```diff
@@ -106,6 +106,6 @@ class="btn-osu-big btn-osu-big--nav-popup js-captcha--submit-button"
 @if (captcha_enabled())
     <script>
         turbolinksReload.load('https://www.recaptcha.net/recaptcha/api.js?render=explicit&onload=initCaptcha&hl={{Lang::getLocale()}}');
-        function initCaptcha() { captcha.init('{{config('captcha.sitekey')}}') }
+        function initCaptcha() { captcha.init('{{config('captcha.sitekey')}}', {{captcha_triggered() ? 'true' : 'false'}}); }
```
[the remark this answers is not in the document]
Comment (nekodex) [reply resources/views/layout/_popup_login.blade.php:109]: Do you have a better idea on how to handle this?

Even if it's updated on logout/whenever, unless we do something like push updates over websocket (overkill?) there's pretty much always going to be edge cases where client-side falls out of sync with the server as to whether or not the captcha should be shown (which is why the `captcha_triggered` return response is a thing...)

The whole point of the captcha is to throttle login attempts, so I feel like an extra required attempt (in certain edge cases) for people with an elevated number of failed logins seems like a fair trade-off?

File: app/Http/Controllers/SessionsController.php [hunk 2]
```diff
@@ -96,6 +96,13 @@ public function store()
                 'user' => Auth::user()->defaultJson(),
             ];
         } else {
+            if (captcha_triggered()) {
+                return response([
+                    'error' => $authError,
+                    'captcha_triggered' => true,
```
[the remark this answers is not in the document]
Comment (nekodex) [reply app/Http/Controllers/SessionsController.php:102]: Do you mean in the wording of the message or something?

[the remark this answers is not in the document]
Comment (nanaya) [reply app/Http/Controllers/SessionsController.php:102]: As in showing the captcha on the login form

File: resources/views/layout/_popup_login.blade.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nanaya) [reply resources/views/layout/_popup_login.blade.php:109]: the `captcha_triggered` isn't included if it's already triggered server side

File: app/Http/Controllers/SessionsController.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (nanaya) [reply app/Http/Controllers/SessionsController.php:102]: Specifically this line here: https://github.com/ppy/osu-web/blob/802d589c964402157549757d78b22997a3a28169/app/Http/Controllers/SessionsController.php#L68

Someone may end up here with front-end still thinking captcha isn't triggered.

File: resources/views/layout/_popup_login.blade.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (nekodex) [reply resources/views/layout/_popup_login.blade.php:109]: So... no, and just fix to show again after logout as required?

[the remark this answers is not in the document]
Comment (nanaya) [reply resources/views/layout/_popup_login.blade.php:109]: I guess

**Review by nanaya — APPROVED** [review]

**Pull request merged**